packages/expo-cli/src/commands/publish/publishAsync.ts

```diff
@@ -150,6 +154,25 @@ function assertValidReleaseChannel(releaseChannel?: string): void {
   }
 }
 
+function assertExpoUpdatesInstalled(projectDir: string, sdkVersion?: string): void {
+  // before sdk 44, expo-update was included in with the expo module
+  if (sdkVersion && semver.lt(sdkVersion, '44.0.0')) {
+    return;
+  }
+
+  const packageJsonPath = resolveFrom.silent(projectDir, './package.json');
+  if (packageJsonPath) {
+    const expoPackageJson = JsonFile.read(packageJsonPath, { json5: true });
+    if (expoPackageJson.dependencies && (expoPackageJson.dependencies as any)['expo-updates']) {
+      return;
+    }
+  }
+
+  throw new Error(
+    `Expo CLI is not installed in this project. Please run "expo install expo-updates".`
```

Comment:
- i believe we should warn and prompt instead of hard fail here to account for one of the possible strange edge cases, so we don't block anybody.
- i suspect you meant to write "expo-updates" and not "Expo CLI"
- i realize now that we don't have a docs page with info about installing expo-updates which accounts for both managed and bare projects. in managed apps people just need to run `expo install expo-updates` but in bare apps we may want to link them to [this docs page](https://docs.expo.dev/bare/installing-updates/)

Reply:
Behavior should be similar to how we ask about installing expo-dev-client (on a managed project only) when eas build kicks off a build with `developmentClient: true`. There's a detection there for managed/bare as well with behavior like you describe.

Reply:
https://github.com/expo/eas-cli/blob/main/packages/eas-cli/src/build/utils/devClient.ts#L16